### DISCUSSION TITLE
How do i stop bacularis from overwriting my config

### DISCUSSION THREAD

**Discussion by mrtn-r** [bacularis/bacularis-app#23] [Q&A] [answered]
Hello,
i referenced an additional config for my bacula instance via @/path/to/config at the end of my /etc/bacula/bacula-dir.conf

When i now use bacularis to create a new  job, this line gets overwritten.
How do i stop bacularis from doing that?

Kind Regards
Martin

**Comment by ganiuszka** [comment]
Hello @mrtn-r

Thank you for writing your question. Yes, it is known problem. It happens because Bacularis reads the Bacula configuration not directly from config files but by the Bacula JSON tools (bdirjson, bfdjson, bsdjson, bbconsjson) and these tools do not support including configuration by ``@/path/to/config``. So, when Bacularis receives this configuration it is already resolved in one big config. Bacularis does not know what config resource comes from where.

I would really like to support config including and I hope it will happen in a day when Bacula JSON tools will start supporting it. Otherwise the only way to do it is to write own Bacula config parser that is something what we would like to avoid.

Best regards,
Marcin Haba (gani)

**Reply by mrtn-r** [reply]
Thanks for your answer, that explains it, i didn't know such tools exist.

Is there some kind of possibility how i can switch bacularis into a "readonly-mode" where it can interact with the config and execute jobs, but all things that require writes are filtered out?

**Comment by ganiuszka** [comment, marked as answer]
Hello Martin,

Yes, I understand your need. There exists solution but not on the Bacularis level but on the system level. You can remove the writable bit from Bacula config files permissions for the web server/PHP user, or just change the ownership to the ``bacula`` user, or ``root`` user or to any other user different than web server/PHP user. The only important is that the web server user and Bacula components can read them.

Alternatively you can use this one line patch that will protect you agains writing to all Bacula config files at once:

```
iff --git a/Common/Modules/ConfigBacula.php b/Common/Modules/ConfigBacula.php
index c839540..2797dbf 100644
--- a/Common/Modules/ConfigBacula.php
+++ b/Common/Modules/ConfigBacula.php
@@ -58,6 +58,7 @@ class ConfigBacula extends CommonModule implements IConfigFormat
         */
        public function write($source, $config)
        {
+               return true;
                $content = $this->prepareConfig($config);
                $orig_umask = umask(0);
                umask(0077);
```

If you installed Bacularis using binary packages, this file you will find in path:

```
/usr/share/bacularis/protected/Common/Modules/ConfigBacula.php
```

Good luck!

Best regards,
Marcin Haba (gani)
